src/Core/Addon/Theme/Theme.php

```diff
@@ -51,11 +51,11 @@ public function __construct(
     ) {
         if (isset($attributes['parent'])) {
             if (null === $configurationCacheDirectory) {
-                $configurationCacheDirectory = (new Configuration())->get('_PS_CACHE_DIR_');
+                $configurationCacheDirectory = _PS_CACHE_DIR_;
```

Comment:
The constant should be injected from the DI to ease testing and reusability.

Reply:
What do you mean about DI ?

Reply:
@FabienPapet if I recall correctly this class is not part of the container, I think we can spare it in this PR

you can see:

```
$theme = new Theme($theme_data);
```

in some places in the core

Reply:
IMHO not worth it for now 👍 it would introduce BC Breaks to fix quite a minor bug. Thanks!

Reply:
I did not see that was not part of  the DI sorry.

I'm trying to remove constants from the core files. I think there should be a way to use it outside of this class. @kpodemski do you think adding a 4th parameter with a default value to  `_PS_CACHE_DIR_` is a good idea ? Should it be done in this PR ?

Reply:
@FabienPapet hmm, maybe that:

`string $configurationCacheDirectory = _PS_CACHE_DIR_`

would be enough, we fix the bug, we still should have the same behavior as most of the core use constant anyway